doc/Installation: make clear the env configs are advanced settings (#16258)
Homebrew/brew: docs/Installation.md

  [Xcode](https://itunes.apple.com/us/app/xcode/id497799835) <sup>[3](#3)</sup>
* The Bourne-again shell for installation (i.e. `bash`) <sup>[4](#4)</sup>

## Advanced Configuration

> The Homebrew installer offers various configuration settings. **Most users can skip this section and instead follow the instructions on the [homepage](https://brew.sh)!**

Reviewer: ```suggestion
The Homebrew installer offers various advanced configuration settings. **Most users can skip this section and instead follow the instructions on the [homepage](https://brew.sh)!**
```

> The Homebrew installer offers various configuration settings. **Most users can skip this section and instead follow the instructions on the [homepage](https://brew.sh)!** If you don't know what these settings mean then you likely don't need to set them as the defaults work great for nearly all users.

Reviewer 2: ```suggestion
The Homebrew installer offers various configuration settings. **Most users can skip this section and instead follow the instructions on the [homepage](https://brew.sh)!**
```

I feel like this line probably isn't necessary since the previous line says that this can be skipped for most users and the next section explicitly proves the git mirrors as solutions to Github.com connection issues.

### Git Remote Mirroring

If you have issues connecting to GitHub.com, you can use Git mirrors for Homebrew's installation and `brew update` by setting `HOMEBREW_BREW_GIT_REMOTE` and/or `HOMEBREW_CORE_GIT_REMOTE` in your shell environment with this script:

```bash
export HOMEBREW_BREW_GIT_REMOTE="..."  # put your Git mirror of Homebrew/brew here
export HOMEBREW_CORE_GIT_REMOTE="..."  # put your Git mirror of Homebrew/homebrew-core here
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
```

> The default Git remote will be used if the corresponding environment variable is unset and works best for most users.

Reviewer: ```suggestion
The default Git remote will be used if the corresponding environment variable is unset and works best for most users.

Note: if you set these variables you are granting these repositories the same level of trust you currently grant to Homebrew itself. You should be extremely confident that these repositories will not be compromised.
```

Reviewer 3: And embolden with `**Note:**`

/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
```

> This will make Homebrew install formulae and casks from the `homebrew/core` and `homebrew/cask` taps using local checkouts of these repositories instead of Homebrew’s API. It is rare you need to enable this setting and it can easily be enabled later after installation should it be necessary.

Reviewer 2: Maybe add something here about how these are usually only necessary if you plan to contribute to one of the core repos.

Author: Better to encourage scoped usage after installation rather than encouraging globally setting it prior to installation IMO.

Reviewer: Agreed.

Reviewer 2: Makes sense

> This will make Homebrew install formulae and casks from the `homebrew/core` and `homebrew/cask` taps using local checkouts of these repositories instead of Homebrew’s API. It is rare you need to enable this setting and it can easily be enabled later after installation should it be necessary.

Reviewer: ```suggestion
This will make Homebrew install formulae and casks from the `homebrew/core` and `homebrew/cask` taps using local checkouts of these repositories instead of Homebrew’s API. Unless you are a Homebrew maintainer or contributor, you should probably not globally enable this setting. It can easily be enabled later after installation should it be necessary.
```
